src/components/CurrentUser/CurrentUser.tsx

```diff
@@ -1,12 +1,59 @@
 import React from 'react';
+import { getUserById } from '../../api/api';
+import { User } from '../../react-app-env';
 import './CurrentUser.scss';
 
-export const CurrentUser: React.FC = () => (
-  <div className="CurrentUser">
-    <h2 className="CurrentUser__title"><span>Selected user: 2</span></h2>
+type Props = {
+  currentUserId: number,
+  clearSelectedUser: () => void,
+};
 
-    <h3 className="CurrentUser__name">Ervin Howell</h3>
-    <p className="CurrentUser__email">Shanna@melissa.tv</p>
-    <p className="CurrentUser__phone">010-692-6593 x09125</p>
-  </div>
-);
+type State = {
+  currentUser: User | null,
+};
+
+export class CurrentUser extends React.Component<Props, State> {
+  state: State = {
+    currentUser: null,
+  };
+
+  componentDidMount() {
+    this.loadUser();
+  }
+
+  componentDidUpdate(prevProps: Props) {
+    if (prevProps.currentUserId !== this.props.currentUserId) {
+      this.loadUser();
+    }
+  }
+
+  loadUser = () => {
+    getUserById(this.props.currentUserId)
+      .then(currentUser => {
```

Comment:
unless a library or legacy codebase forces you to use then/catch, the better choice for readability and maintainability is async/await